.github/workflows/tox.yml

```diff
@@ -6,30 +6,34 @@ on:
 jobs:
   test:
     name: test ${{ matrix.py }} - ${{ matrix.os }}
-    runs-on: ${{ matrix.os }}-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os:
-          - Ubuntu
-          - Windows
-          - MacOs
+          - ubuntu-latest
+          - windows-latest
         py:
           - "3.11"
           - "3.10"
           - "3.9"
           - "3.8"
+        include:
+          - os: macos-latest-xlarge
+            py: "3.10.11"
+          - os: macos-latest-xlarge
+            py: "3.11.8"
     steps:
       - name: Setup python for test ${{ matrix.py }}
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v4
```

Comment:
where does this v4 come from?

Reply:
It was at the time, the most recent version of the [github-built-in setup-python action](https://github.com/actions/setup-python)